File: openapi-test-java-server-spring-client-jdk/testapi/model/Email.java
package testapi.model;

import com.fasterxml.jackson.annotation.JsonValue;

/** Email address wrapper */
public record Email(@JsonValue String value) {
  public Email withValue(String value) {
    return new Email(value);
  };

  @Override
  public String toString() {
    return value;
  };

  /** Converts a String to this type for Spring @PathVariable binding */
  static public Email valueOf(String s) {
    return new Email(s);
  };
}

File: openapi-test-kotlin-server-spring-client-jdk/testapi/model/Email.kt
package testapi.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

/** Email address wrapper */
data class Email @JsonCreator constructor(@get:JsonValue val value: String) {
  override fun toString(): String {
    return value.toString()
  }

  companion object {
    /** Converts a String to this type for Spring @PathVariable binding */
    fun valueOf(s: String): Email = Email(s)
  }
}